src/server/models/grid.js

```diff
@@ -13,46 +16,38 @@ class Grid {
         if (player.trim() === '') throw new Error('Player is empty');
 
 
-        this.players.set(player, this.constructFrameList());
+        this.players.set(player, Player(this.nbFrame));
+        
     }
 
-    constructFrameList() {
-        let frameList = [];
+    calculFrame(namePlayer, mancheNumber) {
 
-        for (let i = 0; i < 10; i++) {
-            frameList.push(new Frame());
-        }
-
-        return frameList;
-    }
-
-    calculFrame(frameTable, mancheNumber) {
-
-        if (mancheNumber == 10) {
-            frameTable[mancheNumber - 1].setScore(calcFrame10(frameTable[mancheNumber - 1]));
+        var frameTable=this.players[namePlayer];
+        if (mancheNumber == this.nbFrame) {
+            frameTable[mancheNumber - 1].setScore(calcFrame10(frameTable[mancheNumber - 1]),this.nbKeel);
```

Comment:
prefer adding a variable for `mancheNumber-1`, maybe on `frameTable[mancheNumber - 1]` ?

Reply:
Done on Jules Branch